server/data/data-jmap/src/main/java/org/apache/james/jmap/api/filtering/impl/FilteringAggregate.java

```diff
@@ -81,6 +83,13 @@ private boolean shouldNotContainDuplicates(List<Rule> rules) {
         return uniqueIdCount == rules.size();
     }
 
+    private boolean shouldHaveTheSameVersionOrExactlyVersionInitialWhenFirstTimeSetRules(Optional<Version> ifInState) {
+        if (ifInState.isPresent() && ifInState.get().equals(Version.INITIAL) && history.getVersionAsJava().isEmpty()) return true;
+        if (history.getVersionAsJava().isPresent() && ifInState.isPresent()
+            && history.getVersionAsJava().get().value() == ifInState.get().getVersion()) return true;
+        return false;
```

Comment:
- it do not pass the checkstylle

 - don't expect me to understand piece of code. even less to maintain it.
 
 
Please write comprehensive code for mere humans, use map filter, etc... Let's write functional code...